core/src/main/scala/com/nvidia/spark/rapids/tool/qualification/Qualification.scala

```diff
@@ -114,9 +117,35 @@ class Qualification(outputDir: String, numRows: Int, hadoopConf: Configuration,
   private def sortForExecutiveSummary(appsSumDesc: Seq[QualificationSummaryInfo],
       order: String): Seq[EstimatedSummaryInfo] = {
     if (QualificationArgs.isOrderAsc(order)) {
-      appsSumDesc.reverse.map(_.estimatedInfo)
+      appsSumDesc.reverse.map(sum =>
+        EstimatedSummaryInfo(
+          sum.estimatedInfo, sum.estimatedFrequency.getOrElse(DEFAULT_JOB_FREQUENCY)))
     } else {
-      appsSumDesc.map(_.estimatedInfo)
+      appsSumDesc.map(sum => EstimatedSummaryInfo(
+        sum.estimatedInfo, sum.estimatedFrequency.getOrElse(DEFAULT_JOB_FREQUENCY)))
+    }
+  }
+
+  // Estimate app frequency based off of all applications in this run, unit jobs per month
+  private def estimateAppFrequency(
+    appsSum: Seq[QualificationSummaryInfo]): Seq[QualificationSummaryInfo] = {
+    val appFrequency = scala.collection.mutable.Map[String, Double]()
+    var windowStart: Long = Long.MaxValue
+    var windowEnd: Long = Long.MinValue
+
+    appsSum.foreach { sum =>
+      appFrequency += (sum.appName -> (1.0 + appFrequency.getOrElse(sum.appName, 0.0)))
+      windowStart = Math.min(sum.startTime, windowStart)
+      windowEnd = Math.max(windowEnd, sum.startTime + sum.estimatedInfo.appDur)
+    }
+    val windowInMonths =
+      if (windowEnd > windowStart) ((windowEnd - windowStart) / (1000.0*60*60*24*30)) else 1.0
+    // Scale frequency to per month, single run jobs are given an estimated frequency of None
+    val monthlyFrequency = appFrequency.map { case (appName, numApps) => (appName ->
+      (if (numApps <= 1) DEFAULT_JOB_FREQUENCY.toDouble else (numApps / windowInMonths)))
+    }
+    appsSum.map { app =>
+      app.copy(estimatedFrequency = Option(monthlyFrequency(app.appName).round))
```

Comment:
be better to getOrElse from monthlyFrequency in case it somehow doesn't have that appName.